src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -130,18 +130,19 @@ trait RegularSync {
           if(error.isDefined){
             val numberBlockFailed = blocks.head.header.number + newBlocks.length
             resumeWithDifferentPeer(peer, reason = s"a block execution error: ${error.get.toString}, in block $numberBlockFailed")
+          } else {
+            headersQueue = headersQueue.drop(blocks.length)
+            if (headersQueue.nonEmpty) {
+              val hashes = headersQueue.take(blockBodiesPerRequest).map(_.hash)
+              waitingForActor = Some(context.actorOf(SyncBlockBodiesRequestHandler.props(peer, hashes)))
+            } else {
+              context.self ! ResumeRegularSync
+            }
           }
         case None =>
           log.error("no total difficulty for latest block") //FIXME: How do we handle this error on our blockchain?
```

Comment:
At least right now we don't recover from this state, so I'll change it to throw an IllegalStateException. There's a task to research into this problem https://iohk.myjetbrains.com/youtrack/issue/EC-165